### ISSUE TITLE
Dependencies are broken. Release 0.2.5.4 (the stable)

### ISSUE THREAD

**Issue by ArkiZh** [DataCanvasIO/HyperGBM#95]
**System information**
- OS Platform and Distribution (e.g., CentOS 7.6): Manjaro
- Python version: 3.10.4
- HyperGBM version: 0.2.5.4
- Other Python packages(run `pip list`):
```
Package               Version
--------------------- ---------
asttokens             2.0.8
backcall              0.2.0
bcrypt                4.0.0
catboost              1.0.6
certifi               2022.6.15
cffi                  1.15.1
charset-normalizer    2.1.1
click                 8.1.3
cloudpickle           2.1.0
convertdate           2.4.0
cryptography          37.0.4
cycler                0.11.0
dask                  2022.7.1
decorator             5.1.1
distributed           2022.7.1
executing             1.0.0
featuretools          1.13.0
fonttools             4.37.1
fsspec                2022.8.2
graphviz              0.20.1
HeapDict              1.0.1
hijri-converter       2.2.4
holidays              0.15
hypergbm              0.2.5.4
hypernets             0.2.5.4
idna                  3.3
imbalanced-learn      0.9.1
ipython               8.4.0
jedi                  0.18.1
Jinja2                3.1.2
joblib                1.1.0
kiwisolver            1.4.4
korean-lunar-calendar 0.2.1
lightgbm              3.3.2
locket                1.0.0
MarkupSafe            2.1.1
matplotlib            3.5.3
matplotlib-inline     0.1.6
msgpack               1.0.4
numpy                 1.23.2
packaging             21.3
pandas                1.4.4
paramiko              2.11.0
parso                 0.8.3
partd                 1.3.0
pexpect               4.8.0
pickleshare           0.7.5
Pillow                9.2.0
pip                   22.1.2
plotly                5.10.0
prettytable           3.4.0
prompt-toolkit        3.0.30
psutil                5.9.1
ptyprocess            0.7.0
pure-eval             0.2.2
pyarrow               9.0.0
pycparser             2.21
Pygments              2.13.0
PyMeeus               0.5.11
PyNaCl                1.5.0
pyparsing             3.0.9
python-dateutil       2.8.2
python-geohash        0.8.5
pytz                  2022.2.1
PyYAML                6.0
requests              2.28.1
scikit-learn          1.1.2
scipy                 1.9.1
seaborn               0.11.2
setuptools            63.4.1
six                   1.16.0
sortedcontainers      2.4.0
stack-data            0.5.0
tblib                 1.7.0
tenacity              8.0.1
threadpoolctl         3.1.0
toolz                 0.12.0
tornado               6.1
tqdm                  4.64.0
traitlets             5.3.0
urllib3               1.26.12
wcwidth               0.2.5
wheel                 0.37.1
woodwork              0.18.0
xgboost               1.6.2
XlsxWriter            3.0.3
zict                  2.2.0

```

**Describe the current behavior**
Failed to run a simple demo.

**Describe the expected behavior**

**Standalone code to reproduce the issue**

Environment:
```
conda clean -a
pip cache purge
conda create -n gbm
conda activate gbm
conda install pip
pip install hypergbm

```
Log:
`Successfully installed MarkupSafe-2.1.1 XlsxWriter-3.0.3 asttokens-2.0.8 backcall-0.2.0 bcrypt-4.0.0 catboost-1.0.6 cffi-1.15.1 charset-normalizer-2.1.1 click-8.1.3 cloudpickle-2.1.0 convertdate-2.4.0 cryptography-37.0.4 cycler-0.11.0 dask-2022.7.1 decorator-5.1.1 distributed-2022.7.1 executing-1.0.0 featuretools-1.13.0 fonttools-4.37.1 fsspec-2022.8.2 graphviz-0.20.1 heapdict-1.0.1 hijri-converter-2.2.4 holidays-0.15 hypergbm-0.2.5.4 hypernets-0.2.5.4 idna-3.3 imbalanced-learn-0.9.1 ipython-8.4.0 jedi-0.18.1 jinja2-3.1.2 joblib-1.1.0 kiwisolver-1.4.4 korean-lunar-calendar-0.2.1 lightgbm-3.3.2 locket-1.0.0 matplotlib-3.5.3 matplotlib-inline-0.1.6 msgpack-1.0.4 numpy-1.23.2 packaging-21.3 pandas-1.4.4 paramiko-2.11.0 parso-0.8.3 partd-1.3.0 pexpect-4.8.0 pickleshare-0.7.5 pillow-9.2.0 plotly-5.10.0 prettytable-3.4.0 prompt-toolkit-3.0.30 psutil-5.9.1 ptyprocess-0.7.0 pure-eval-0.2.2 pyarrow-9.0.0 pycparser-2.21 pygments-2.13.0 pymeeus-0.5.11 pynacl-1.5.0 pyparsing-3.0.9 python-dateutil-2.8.2 python-geohash-0.8.5 pytz-2022.2.1 pyyaml-6.0 requests-2.28.1 scikit-learn-1.1.2 scipy-1.9.1 seaborn-0.11.2 six-1.16.0 sortedcontainers-2.4.0 stack-data-0.5.0 tblib-1.7.0 tenacity-8.0.1 threadpoolctl-3.1.0 toolz-0.12.0 tornado-6.1 tqdm-4.64.0 traitlets-5.3.0 urllib3-1.26.12 wcwidth-0.2.5 woodwork-0.18.0 xgboost-1.6.2 zict-2.2.0
`

Code sample:

```
import pandas as pd
from sklearn import datasets
from sklearn.model_selection import train_test_split

X,y = datasets.load_breast_cancer(as_frame=True,return_X_y=True)
X_train,X_test,y_train,y_test = train_test_split(X,y,train_size=0.7,random_state=335)
train_data = pd.concat([X_train,y_train],axis=1)

from hypergbm import make_experiment


experiment = make_experiment(train_data, target='target', reward_metric='precision')
estimator = experiment.run()
```

The output:
```
09-01 14:28:01 E hypernets.m.hyper_model.py 71 - run_trail failed! trail_no=7
09-01 14:28:01 E hypernets.m.hyper_model.py 73 - Traceback (most recent call last):
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/hypernets/model/hyper_model.py", line 61, in _run_trial
    scores, oof, oof_scores = estimator.fit_cross_validation(X, y, stratified=True, num_folds=num_folds,
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/hypergbm/hyper_gbm.py", line 312, in fit_cross_validation
    fold_est.fit(x_train_fold, y_train_fold, **fit_kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/hypergbm/estimators.py", line 482, in fit
    return self.fit_with_encoder(super().fit, X, y, kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/hypergbm/estimators.py", line 181, in fit_with_encoder
    return fn_fit(X, y, **kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/core.py", line 575, in inner_f
    return f(**kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/sklearn.py", line 1400, in fit
    self._Booster = train(
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/core.py", line 575, in inner_f
    return f(**kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/training.py", line 159, in train
    _assert_new_callback(callbacks)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/training.py", line 25, in _assert_new_callback
    raise ValueError(
ValueError: Old style callback was removed in version 1.6.  See: https://xgboost.readthedocs.io/en/latest/python/callbacks.html.

09-01 14:28:01 E hypernets.m.hyper_model.py 71 - run_trail failed! trail_no=8
09-01 14:28:01 E hypernets.m.hyper_model.py 73 - Traceback (most recent call last):
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/hypernets/model/hyper_model.py", line 61, in _run_trial
    scores, oof, oof_scores = estimator.fit_cross_validation(X, y, stratified=True, num_folds=num_folds,
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/hypergbm/hyper_gbm.py", line 312, in fit_cross_validation
    fold_est.fit(x_train_fold, y_train_fold, **fit_kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/hypergbm/estimators.py", line 482, in fit
    return self.fit_with_encoder(super().fit, X, y, kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/hypergbm/estimators.py", line 181, in fit_with_encoder
    return fn_fit(X, y, **kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/core.py", line 575, in inner_f
    return f(**kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/sklearn.py", line 1400, in fit
    self._Booster = train(
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/core.py", line 575, in inner_f
    return f(**kwargs)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/training.py", line 159, in train
    _assert_new_callback(callbacks)
  File "~/miniconda3/envs/gbm/lib/python3.10/site-packages/xgboost/training.py", line 25, in _assert_new_callback
    raise ValueError(
ValueError: Old style callback was removed in version 1.6.  See: https://xgboost.readthedocs.io/en/latest/python/callbacks.html.
```

**Comment by lixfz** [comment]
We fixed this issue in version 0.2.5.5，upgrade HyperGBM and try again please .

**Comment by ArkiZh** [comment]
> We fixed this issue in version 0.2.5.5，upgrade HyperGBM and try again please .

Works fine, thanks🙂
